ChangeLog

```diff
@@ -35,17 +35,31 @@ MODS:
 * Added bonus updaters for hero specialties
 * Added allOf, anyOf and noneOf qualifiers for bonus limiters
 * Added bonus limiters: alignment, faction and terrain
+* Supported new terrains, new battlefields, custom water and rock terrains

```

Comment:
* Following special buildings becomes available in the fan towns:
** attackVisitingBonus
** defenceVisitingBonus
** spellPowerVisitingBonus
** knowledgeVisitingBonus
** experienceVisitingBonus
** lighthouse
** treasury